x/halo/keeper/keeper.go

```diff
@@ -123,7 +122,10 @@ func (k *Keeper) mintCoins(ctx sdk.Context, recipient sdk.AccAddress, coins sdk.
 
 // depositFor is an internal helper function to deposit.
 func (k *Keeper) depositFor(ctx sdk.Context, signer sdk.AccAddress, recipient sdk.AccAddress, underlying sdk.Int) (amount sdk.Int, err error) {
-	round, _ := k.GetRound(ctx, k.GetLastRoundId(ctx))
+	round, found := k.GetRound(ctx, k.GetLastRoundId(ctx))
+	if !found {
+		return sdk.Int{}, fmt.Errorf("round %d not found", k.GetLastRoundId(ctx))
```

Comment:
Suggestion here, we put the response of `GetLastRoundId` into a variable so we don't need to query it twice in the scenario there is an error? In theory this should never happen but would clean up state reads.